assignments/assignment7/faulty-oops.md

```diff
@@ -0,0 +1,65 @@
+# AESD Assignment 7 faulty-oops write-up
+
+# Description
+### Command:
+        $ echo “hello_world” > /dev/faulty
+
+### Output:
+<img src="A7-faulty-oops.png" alt="A7-faulty-oops" title="A7-faulty-oops"> 
+
+This is called a oops message.
+
+### Analysis:
+
+        Unable to handle kernel NULL pointer dereference at virtual address 0000000000000000
+
+This line tells that the kernel was dereferencing a NULL pointer. This causes a fault and kills all the current tasks and may leak kernel resources. The processor signals a page fault to the OS when dereferencing a invalid pointer.
+
+        CPU: 0 PID: 158 Comm: sh Tainted: G           O      5.15.18 #1
+
+This line tells the core running the process that caused the fault. PID 158 is the process ID of the program that caused the fault. "Tainted: G" indicates that a proprietary module was loaded.
+
+        pc : faulty_write+0x14/0x20 [faulty]
+
+This lines gives information about the function that caused the fault, i.e, faulty_write function and the module containing the function, i.e, faulty.
```

Comment:
@ameyflash it also shows the location within `faulty_write` where the fault occurred, 0x14 bytes into a function of length 0x20 bytes.